widgetssdk/src/main/java/com/glia/widgets/core/queue/QueueRepository.kt

```diff
@@ -32,57 +40,65 @@ internal class QueueRepositoryImpl(private val gliaCore: GliaCore, private val c
         .doOnSubscribe { fetchQueues() }
         .distinctUntilChanged()
 
-    override val relevantQueueIds: List<String>
-        get() = _queuesState.value
-            ?.let { it as? QueuesState.Queues }
-            ?.run { queues.map { it.id } }
-            .orEmpty()
+    private val _relevantQueueIds: Flowable<List<String>>
+        get() = queuesState
+            .filter { it !is QueuesState.Loading }
+            .map { it.queuesOrEmpty() }
+            .map { getIds(it) }
 
-    init {
-        subscribeToQueues()
-        subscribeToQueueUpdates()
+    override val relevantQueueIds: Single<List<String>>
+        get() = _relevantQueueIds.firstOrError()
+
+    override fun initialize() {
+        fetchQueues()
     }
 
     private fun fetchQueues() {
-        if (siteQueues.value?.isSuccess != true) {
+        if (siteQueues.value?.isSuccess != true && gliaCore.isInitialized) {
             _queuesState.onNext(QueuesState.Loading)
+
             gliaCore.getQueues { queues, exception ->
-                when {
-                    queues != null -> siteQueues.onNext(Result.success(queues.toList()))
-                    else -> {
-                        siteQueues.onNext(Result.failure(exception ?: RuntimeException("Fetching queues failed")))
-                        Logger.e(TAG, "Fetching queues failed", exception)
-                    }
-                }
+                queues?.run { siteQueuesReceived(toList()) } ?: getSiteQueuesError(exception)
             }
         }
     }
 
+    private fun siteQueuesReceived(queues: List<Queue>) {
+        siteQueues.onNext(Result.success(queues))
+        subscribeToQueues()
+        subscribeToQueueUpdates()
+    }
+
+    private fun getSiteQueuesError(exception: GliaException?) {
+        siteQueues.onNext(Result.failure(exception ?: RuntimeException("Fetching queues failed")))
+        Logger.e(TAG, "Fetching queues failed", exception)
+    }
+
+    private fun getIds(queues: List<Queue>): List<String> = queues.map { it.id }
```

Comment:
NIT. It's a tiny function, I think `queues.map { it.id }` can just be put without having a function.